utilities/artifactory.py

```
@@ -161,19 +206,56 @@ def cleanup_artifactory_secret_and_config_map(
+        client: Kubernetes dynamic client used to create and delete the resources.
+        create_secret: Whether to create/retrieve the Artifactory Secret. Defaults to True.
+        create_config_map: Whether to create/retrieve the Artifactory ConfigMap. Defaults to True.
+
+    Yields:
+        ArtifactoryCredentials: Created Secret and/or ConfigMap for DV/source auth.
+
+    Raises:
+        ValueError: If both ``create_secret`` and ``create_config_map`` are False.
+    """
+    if not create_secret and not create_config_map:
+        raise ValueError("At least one of create_secret or create_config_map must be True")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)